### PR TITLE
vite migration

### PR DESCRIPTION
**Pull request by sainak**
## Proposed Changes

- migrate build tools to vite
- upgraded storybook and stories 

## TODO

- [x] verify that all components work as expected in a production build
- [x] fix service worker (checkout vite-pwa)
- [x] fix ci issues
- [x] fix storybook
- [x] fix generated css file size 

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

**Comment by sainak** [comment]
@rithviknishad the components contaning the phone number field break while previewing a production build, I'm not able to pin point the actual issue but here's the error that I got 
```
Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: [missing argument].[missing argument]
```

image


please check this issue


Edit: fixed it

**Comment by sainak** [comment]
@rithviknishad the issue is fixed now.
We will have to be careful with what external libraries we use.

**Comment by gigincg** [comment]
@sainak Can you update the `netlify.toml` and the `Dockerfile` so that @mathew-alex Can test this out in those environments?

cc: @rithviknishad

**Comment by sainak** [comment]
@gigincg changes to deployment config won't be needed as internally npm run build is calling vite
You can see the latest deploy preview, the components are working fine now but the css file is quite large as compared to cra builds, service worker might be broken, and I'm working on fixing the storybook

**Comment by shivankacker** [comment]
Just to quantify the impact of this PR, starting the current **CRA app takes 1 minute and 36 seconds** on my laptop with 16 gb ram and an i5-11400H @ 2.70GHz  2.69 GHz CP.
The **Vite app takes just 40 seconds to start** which makes the app start **215% faster** than before. Hot reload also takes considerably less time than before. This allows CARE to be developed in less powerful computers. Great feat @sainak

**Comment by sainak** [comment]
@rithviknishad please update the ci for storybook the build commnad is `build:storybook` and the dist folder is `build-storybook`
also can you verify if the service worker is working correctly?

**Comment by Ashesh3** [comment]
With this my build time goes down from `210s` to `57s` and hot reload time goes down to `6s` from `28s`. This is really helpful. Great work! 
Tested on 3.0 GHz Ampere® Altra™ x4 core CPU with 32GB memory.

File: .storybook/main.ts [hunk 1]
```diff
@@ -1,21 +1,19 @@
-import type { StorybookConfig } from "@storybook/react-vite";
-
-const config: StorybookConfig = {
```
Comment (sainak) [inline .storybook/main.ts]: lets keep the StorybookConfig type as it'll help us for type hints

File: .storybook/main.ts [hunk 2]
```diff
@@ -1,21 +1,19 @@
...
     "@storybook/addon-essentials",
     "@storybook/addon-interactions",
-  ],
-  core: {
-    builder: {
-      name: "@storybook/builder-vite",
+    {
+      name: "@storybook/addon-postcss",
       options: {
-        viteConfigPath: "./.storybook/vite.config.ts",
```
Comment (sainak) [inline .storybook/main.ts]: I would recommend using the vite builder as we are using it for our project
and vite builder requires its own config that i created under .storybook/vite.config.ts

File: .storybook/main.ts [hunk 3]
```diff
@@ -1,21 +1,19 @@
...
+        postcssLoaderOptions: {
+          implementation: require("postcss"),
+        },
       },
     },
-  },
-  staticDirs: ["../public"],
```
Comment (sainak) [inline .storybook/main.ts]: this is required

File: package.json [hunk 4]
```diff
@@ -47,8 +47,8 @@
     "lint-fix": "eslint ./src --fix",
     "format": "prettier ./src --write",
     "dev:storybook": "storybook dev -p 6006 ",
-    "build:storybook": "storybook build -o build-storybook",
-    "preview:storybook": "vite preview --outDir build-storybook"
+    "build:storybook": "storybook build -o storybook build",
+    "preview:storybook": "vite preview --outDir storybook build"
```
Comment (sainak) [inline package.json]: this is a conflict with the project build path

File: package.json [hunk 5]
```diff
@@ -115,9 +115,9 @@
     "@storybook/addon-essentials": "^7.0.0-rc.6",
     "@storybook/addon-interactions": "^7.0.0-rc.6",
     "@storybook/addon-links": "^7.0.0-rc.6",
+    "@storybook/addon-mdx-gfm": "^7.0.0-rc.6",
     "@storybook/addon-postcss": "^2.0.0",
     "@storybook/blocks": "^7.0.0-rc.6",
-    "@storybook/builder-vite": "^7.0.0-rc.6",
```
Comment (sainak) [inline package.json]: this is needed

**Comment by sainak** [comment]
@gigincg @mathew-alex  the PR is ready to be merged in develop with one breaking change that is the env variable `PUBLIC_URL` needs to be renamed to `REACT_PUBLIC_URL`

**Review by skks1212 — APPROVED** [review]
LGTM!

**Comment by khavinshankar** [comment]
@sainak can you fix the merge conflicts

File: src/service-worker.ts [hunk 6]
```diff
@@ -1,138 +0,0 @@
...
-    } else {
-      event.waitUntil(
-        self.registration.showNotification("Care - CoronaSafe Network", {
-          body: data.title,
-          tag: data.external_id,
-        })
-      );
-    }
-  }
-});
```
Comment (khavinshankar) [inline src/service-worker.ts]: @sainak is this handled anywhere else? I don't see this reimplemented

**Review by khavinshankar — CHANGES_REQUESTED** [review]

File: src/service-worker.ts (hunk 6 above)
Comment (sainak) [reply src/service-worker.ts]: yes, it is handled by vite-pwa

**Review by khavinshankar — APPROVED** [review]

**Review by gigincg — APPROVED** [review]

**Pull request merged**